subsys/net/lib/nrf_cloud/src/nrf_cloud_rest.c

```diff
@@ -131,29 +131,68 @@ static const char *const agps_req_type_strings[] = {
 	[NRF_CLOUD_REST_AGPS_REQ_CUSTOM]	= AGPS_REQ_TYPE_STR_CUSTOM,
 };
 
+/* Generate an authorization header value string in the form:
+ * "Authorization: Bearer JWT \r\n"
+ */
 static int generate_auth_header(const char *const tok, char **auth_hdr_out)
 {
-	if (!tok || !auth_hdr_out) {
+	if (!auth_hdr_out) {
+		LOG_ERR("Cannot generate auth header, no output pointer given.");
 		return -EINVAL;
 	}
 
-	int ret;
-	size_t buff_size = sizeof(AUTH_HDR_BEARER_TEMPLATE) + strlen(tok);
+	/* These lengths NOT including null terminators */
+	int tok_len = tok ? strlen(tok) : (CONFIG_MODEM_JWT_MAX_LEN - 1);
```

Comment:
if
```
CONFIG_MODEM_JWT=n
CONFIG_NRF_CLOUD_REST_AUTOGEN_JWT=n
```
a build error will result from `CONFIG_MODEM_JWT_MAX_LEN`

Reply:
Fixed, thank you